chia/full_node/full_node.py

```diff
@@ -1601,44 +1601,45 @@ async def respond_transaction(
         if self.sync_store.get_sync_mode():
             status = MempoolInclusionStatus.FAILED
             error: Optional[Err] = Err.NO_TRANSACTIONS_WHILE_SYNCING
+            self.mempool_manager.remove_seen(spend_name)
         else:
             try:
                 cost_result = await self.mempool_manager.pre_validate_spendbundle(transaction)
             except Exception as e:
                 self.mempool_manager.remove_seen(spend_name)
                 raise e
-            async with self.blockchain.lock:
+            async with self.mempool_manager.lock:
```

Comment:
`add_spendbundle()` verifies removals against the `coin_store`, which I assume may not change under this functions feet. What if there's a reorg during this validation?

Reply:
If there is a reorg, then new_peak will get called, and the mempool will get cleared